src/Handler/Table/Preview/PreviewTableHandler.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\StorageDriver\Teradata\Handler\Table\Preview;
+
+use Google\Protobuf\Internal\GPBType;
+use Google\Protobuf\Internal\Message;
+use Google\Protobuf\Internal\RepeatedField;
+use Google\Protobuf\NullValue;
+use Google\Protobuf\Value;
+use Keboola\Datatype\Definition\BaseType;
+use Keboola\Datatype\Definition\Teradata;
+use Keboola\StorageDriver\Command\Table\PreviewTableCommand;
+use Keboola\StorageDriver\Command\Table\PreviewTableResponse;
+use Keboola\StorageDriver\Contract\Driver\Command\DriverCommandHandlerInterface;
+use Keboola\StorageDriver\Credentials\GenericBackendCredentials;
+use Keboola\StorageDriver\Teradata\TeradataSessionManager;
+use Keboola\TableBackendUtils\Column\Teradata\TeradataColumn;
+use Keboola\TableBackendUtils\Escaping\Teradata\TeradataQuote;
+use Keboola\TableBackendUtils\Table\Teradata\TeradataTableReflection;
+
+class PreviewTableHandler implements DriverCommandHandlerInterface
+{
+    public const STRING_MAX_LENGTH = 50;
+
+    private TeradataSessionManager $manager;
+
+    public function __construct(TeradataSessionManager $manager)
+    {
+        $this->manager = $manager;
+    }
+
+    /**
+     * @inheritDoc
+     * @param GenericBackendCredentials $credentials
+     * @param PreviewTableCommand $command
+     */
+    public function __invoke(
+        Message $credentials,
+        Message $command,
+        array $features
+    ): ?Message {
+        assert($credentials instanceof GenericBackendCredentials);
+        assert($command instanceof PreviewTableCommand);
+
+        // validate
+        assert($command->getPath()->count() === 1, 'CreateTableCommand.path is required and size must equal 1');
+        assert(!empty($command->getTableName()), 'CreateTableCommand.tableName is required');
+        assert($command->getColumns()->count() > 0, 'CreateTableCommand.columns is required');
+
+        try {
+            $db = $this->manager->createSession($credentials);
+            /** @var string $databaseName */
+            $databaseName = $command->getPath()[0];
+
+            // build sql
+            $columns = $this->repeatedStringToArray($command->getColumns());
+            // TODO changeSince, changeUntil
+            // TODO fulltextSearch
+            // TODO whereFilters
+            $selectTableSql = sprintf(
+                "SELECT %s %s\nFROM %s.%s",
+                $command->getLimit() ? sprintf('TOP %d', $command->getLimit()) : '',
+                implode(', ', array_map([TeradataQuote::class, 'quoteSingleIdentifier'], $columns)),
+                TeradataQuote::quoteSingleIdentifier($databaseName),
+                TeradataQuote::quoteSingleIdentifier($command->getTableName())
+            );
+
+            if ($command->hasOrderBy() && $command->getOrderBy()) {
+                $orderBy = $command->getOrderBy();
+                $selectTableSql .= sprintf(
```

Comment:
osobne bych radeji videl na to nejakej query builder. 
Prvne to jde unit-testovat coz krome samotneho testu zlepsi udrzitelnost kodu (proste vidis jakou query to generuje)
Druhak kdyz se podivas do Connection, tak ta query se pak pouziva jeste pro exportToFile (pres toto volani https://github.com/keboola/connection/blob/28b98d4255596eecea0198cb93b8c54c9470c02f/legacy-app/application/modules/storage/services/BucketBackend/Exasol.php#L716 )

Takze na to prosim udelej nejakej query builder (to chci pak otevrit tema kde by to melo realne byt, ale zatim nejakej udelej) a dej to tam. Plus testy prosim. Dekuji :)

Reply:
stejne tak bych se asi trochu inspiroval u toho stareho query builderu a trochu to buildovani rozhazel do metod

Reply:
BTW asi by bylo fajn nereinventovat kola https://www.doctrine-project.org/projects/doctrine-dbal/en/latest/reference/query-builder.html

Reply:
Dohodli jsme se, ze nebudeme zatim delat QueryBuilder.
BTW Doctrine nepodporuje Teradatu.